test/moments.jl

```diff
@@ -1,122 +1,284 @@
 using StatsBase
 using Base.Test
 
-##### weighted var & std
-
-x = rand(10)
-wv = weights(rand(10))
-m = mean(x, wv)
-
-@test var(x, wv)           ≈ sum(abs2.(x .- m), wv) ./ sum(wv)
-@test var(x, wv; mean=0)   ≈ sum(abs2.(x), wv) ./ sum(wv)
-@test var(x, wv; mean=1.0) ≈ sum(abs2.(x .- 1.0), wv) ./ sum(wv)
-
-@test std(x, wv)           ≈ sqrt(var(x, wv))
-@test std(x, wv; mean=0)   ≈ sqrt(var(x, wv; mean=0))
-@test std(x, wv; mean=1.0) ≈ sqrt(var(x, wv; mean=1.0))
-
-(m, v) = mean_and_var(x)
-@test m == mean(x)
-@test v == var(x)
-
-(m, s) = mean_and_std(x)
-@test m == mean(x)
-@test s == std(x)
-
-(m, v) = mean_and_var(x, wv)
-@test m == mean(x, wv)
-@test v == var(x, wv)
-
-(m, s) = mean_and_std(x, wv)
-@test m == mean(x, wv)
-@test s == std(x, wv)
-
-x = rand(5, 6)
-w1 = rand(5)
-w2 = rand(6)
-wv1 = weights(w1)
-wv2 = weights(w2)
-m1 = mean(x, wv1, 1)
-m2 = mean(x, wv2, 2)
-
-@test var(x, wv1, 1; mean=0) ≈ sum(abs2.(x) .* w1, 1) ./ sum(wv1)
-@test var(x, wv2, 2; mean=0) ≈ sum(abs2.(x) .* w2', 2) ./ sum(wv2)
-
-@test var(x, wv1, 1; mean=m1) ≈ sum(abs2.(x .- m1) .* w1, 1) ./ sum(wv1)
-@test var(x, wv2, 2; mean=m2) ≈ sum(abs2.(x .- m2) .* w2', 2) ./ sum(wv2)
-
-@test var(x, wv1, 1) ≈ sum(abs2.(x .- m1) .* w1, 1) ./ sum(wv1)
-@test var(x, wv2, 2) ≈ sum(abs2.(x .- m2) .* w2', 2) ./ sum(wv2)
-
-@test std(x, wv1, 1)          ≈ sqrt.(var(x, wv1, 1))
-@test std(x, wv2, 2)          ≈ sqrt.(var(x, wv2, 2))
-@test std(x, wv1, 1; mean=0)  ≈ sqrt.(var(x, wv1, 1; mean=0))
-@test std(x, wv2, 2; mean=0)  ≈ sqrt.(var(x, wv2, 2; mean=0))
-@test std(x, wv1, 1; mean=m1) ≈ sqrt.(var(x, wv1, 1; mean=m1))
-@test std(x, wv2, 2; mean=m2) ≈ sqrt.(var(x, wv2, 2; mean=m2))
-
-for d in 1:2
-    (m, v) = mean_and_var(x, d)
-    @test m == mean(x, d)
-    @test v == var(x, d)
-
-    (m, s) = mean_and_std(x, d)
-    @test m == mean(x, d)
-    @test s == std(x, d)
+@testset "StatsBase.Moments" begin
+weight_funcs = (weights, aweights, fweights, pweights)
+
+@testset "Variance and Standard Deviation" begin
+    @testset "Vectors" begin
+        x = [0.57, 0.10, 0.91, 0.72, 0.46]
```

Comment:
Can you add an observation with a `0` weight to test their handling? Shouldn't change the result.